easybuild/easyblocks/c/cfdemcoupling.py

```diff
@@ -0,0 +1,169 @@
+
+        env.setvar('CFDEM_VERSION', self.version)
+        env.setvar('CFDEM_PROJECT_DIR', self.cfdem_project_dir)
+
+        # define $CFDEM_PROJECT_USER_DIR to an empty existing directory
+        project_user_dir = os.path.join(self.builddir, 'project_user_dir')
+        env.setvar('CFDEM_PROJECT_USER_DIR', project_user_dir)
+        mkdir(project_user_dir, parents=True)
+
+        cfdem_bashrc = os.path.join(self.cfdem_project_dir, 'src', 'lagrangian', 'cfdemParticle', 'etc', 'bashrc')
+        env.setvar('CFDEM_bashrc', cfdem_bashrc)
+
+        env.setvar('CFDEM_LIGGGHTS_SRC_DIR', os.path.join(self.liggghts_dir, 'src'))
+        env.setvar('CFDEM_LIGGGHTS_MAKEFILE_NAME', 'auto')
+
+        lpp_dirs = glob.glob(os.path.join(self.builddir, 'LPP-*'))
+        if len(lpp_dirs) == 1:
+            env.setvar('CFDEM_LPP_DIR', lpp_dirs[0])
+        else:
+            raise EasyBuildError("Failed to isolate LPP-* directory in %s", self.builddir)
+
+        # build in parallel
+        env.setvar("WM_NCOMPPROCS", str(self.cfg['parallel']))
+
+        vtk_root = get_software_root('VTK')
+        if vtk_root:
+            vtk_ver_maj_min = '.'.join(get_software_version('VTK').split('.')[:2])
+            vtk_inc = os.path.join(vtk_root, 'include', 'vtk-%s' % vtk_ver_maj_min)
+            if os.path.exists(vtk_inc):
+                env.setvar('VTK_INC_USR', '-I%s' % vtk_inc)
+            else:
+                raise EasyBuildError("Expected directory %s does not exist!", vtk_inc)
+
+            vtk_lib = os.path.join(vtk_root, 'lib')
+            if os.path.exists(vtk_lib):
+                env.setvar('VTK_LIB_USR', '-L%s' % vtk_lib)
+            else:
+                raise EasyBuildError("Expected directory %s does not exist!", vtk_lib)
+        else:
+            raise EasyBuildError("VTK not included as dependency")
+
+        # can't seem to use defined 'cfdemSysTest' alias, so call cfdemSystemTest.sh script directly...
+        cmd = "source $CFDEM_bashrc && $CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/cfdemSystemTest.sh"
+        run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+
+    def build_step(self):
+        """Custom build procedure for CFDEMcoupling."""
+
+        if get_software_root('OpenFOAM'):
+            openfoam_ver = get_software_version('OpenFOAM')
+            openfoam_maj_ver = openfoam_ver.split('.')[0]
+        else:
+            raise EasyBuildError("OpenFOAM not included as dependency")
+
+        # make sure expected additionalLibs_* file is available
+        addlibs_subdir = os.path.join('src', 'lagrangian', 'cfdemParticle', 'etc', 'addLibs_universal')
+        src_addlibs = os.path.join(self.cfdem_project_dir, addlibs_subdir, 'additionalLibs_%s.x' % openfoam_maj_ver)
+        target_addlibs = os.path.join(self.cfdem_project_dir, addlibs_subdir, 'additionalLibs_%s' % openfoam_ver)
+        copy_file(src_addlibs, target_addlibs)
+
+        # can't seem to use defined 'cfdemCompCFDEMall' alias...
```

Comment:
Should this comment be similar as the one above or ?
I'm fine with it as it is, just checking if you missed it or left it on purpose.

Reply:
I missed it, but do we need to clarify it in both places? I'd say it's fine as is